packages/vite-plugin-cloudflare/src/plugin-config.ts

```diff
@@ -11,26 +11,40 @@ import type {
 
 export type PersistState = boolean | { path: string };
 
-interface PluginWorkerConfig {
-	configPath: string;
+interface BaseWorkerConfig {
 	viteEnvironment?: { name?: string };
 }
 
-export interface PluginConfig extends Partial<PluginWorkerConfig> {
-	auxiliaryWorkers?: PluginWorkerConfig[];
+interface EntryWorkerConfig extends BaseWorkerConfig {
+	configPath?: string;
+}
+
+interface AuxiliaryWorkerConfig extends BaseWorkerConfig {
+	configPath: string;
+}
+
+export interface PluginConfig extends EntryWorkerConfig {
+	auxiliaryWorkers?: AuxiliaryWorkerConfig[];
 	persistState?: PersistState;
 }
 
 type Defined<T> = Exclude<T, undefined>;
 
-export type AssetsOnlyConfig = SanitizedWorkerConfig & {
-	assets: Defined<SanitizedWorkerConfig['assets']>;
-};
-
-export type WorkerConfig = SanitizedWorkerConfig & {
+interface BaseConfig {
+	topLevelName: Defined<SanitizedWorkerConfig['topLevelName']>;
 	name: Defined<SanitizedWorkerConfig['name']>;
-	main: Defined<SanitizedWorkerConfig['main']>;
-};
+	compatibility_date: Defined<SanitizedWorkerConfig['compatibility_date']>;
+}
+
+export type AssetsOnlyConfig = SanitizedWorkerConfig &
```

Comment:
If you only have a single parent then I think you could use inheritance here?

```suggestion
export interface AssetsOnlyConfig extends BaseWorkerConfig {
```

Reply:
[595081b](https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/127/commits/595081b39139b5e0c20fc3579c1d7110a89de655)